### PR TITLE
[WA-2]

### PR DESCRIPTION
**Pull request by plopdark**

### CODE REVIEW

File: src/app/pages/weather-main/weather-main.component.scss [hunk 1]
```diff
@@ -224,37 +223,37 @@
 }
 
 @media (max-width: 1920px) {
```
Comment (Oskar2301) [inline src/app/pages/weather-main/weather-main.component.scss:225]: read about mixin in scss and create mixins for media

File: src/app/pages/weather-page/weather-page.component.scss [hunk 2]
```diff
@@ -0,0 +1,373 @@
...
+        rgba(0, 0, 0, 0.03) -1.68%,
+        rgba(255, 255, 255, 0.09) 127.54%,
+        rgba(255, 255, 255, 0.11) 127.56%,
+        rgba(255, 255, 255, 0.11) 127.57%,
+        rgba(255, 255, 255, 0.15) 127.58%
+      );
+    }
+  }
+
+  //   ------------------------------------------------------
```
Comment (Oskar2301) [inline src/app/pages/weather-page/weather-page.component.scss]: ?

File: src/app/pages/weather-page/weather-page.component.scss [hunk 3]
```diff
@@ -0,0 +1,373 @@
...
+      display: flex;
+      flex-direction: row;
+      align-items: flex-end;
+    }
+  }
+  &__main-date {
+    font-size: 24px;
+    margin-bottom: 20px;
+  }
+  //   --------------------------------
```
Comment (Oskar2301) [inline src/app/pages/weather-page/weather-page.component.scss]: ?

File: src/app/pages/weather-page/weather-page.component.scss [hunk 4]
```diff
@@ -0,0 +1,373 @@
...
+  &__vectors {
+    margin-top: 30px;
+    display: flex;
+    flex-direction: row;
+    align-items: center;
+    justify-content: space-between;
+  }
+}
+
+// ---------------------------------------------------
```
Comment (Oskar2301) [inline src/app/pages/weather-page/weather-page.component.scss]: ?

File: src/app/pages/weather-page/weather-page.component.scss [hunk 5]
```diff
@@ -0,0 +1,373 @@
...
+
+      &_indicators {
+        gap: 13px;
+        margin-right: 10px;
+      }
+    }
+  }
+}
+
+@media (max-width: 1190px) {
```
Comment (Oskar2301) [inline src/app/pages/weather-page/weather-page.component.scss]: use mixin

File: src/app/shared/components/day-part/day-part.component.scss [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+  }
+
+  &__params {
+    font-size: 18px;
+    font-weight: 400;
+    margin-bottom: 0;
+  }
+}
+
+// --------------------------------------
```
Comment (Oskar2301) [inline src/app/shared/components/day-part/day-part.component.scss]: delete this comment in all files scss

File: src/app/shared/components/day-part/day-part.component.scss [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+      gap: 13px;
+    }
+
+    &__img {
+      width: 45px;
+    }
+  }
+}
+
+@media (max-width: 1190px) {
```
Comment (Oskar2301) [inline src/app/shared/components/day-part/day-part.component.scss]: mixin

File: src/app/shared/components/day-part/day-part.component.ts [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+  @Input() partBlock: IDayPart;
+  public currentFirstHour: any;
+  public currentSecondHour: any;
+  public currentFirstImage: string = '';
+  public currentSecondImage: string = '';
+
+  constructor(public weatherService: WeatherService) {}
+
+  ngOnInit(): void {
+    this.weatherService.triggerRequest.pipe().subscribe(() => {
```
Comment (Oskar2301) [inline src/app/shared/components/day-part/day-part.component.ts]: unsubscribe

File: src/app/shared/components/next-days/next-days.component.scss [hunk 9]
```diff
@@ -0,0 +1,99 @@
+.weather-page {
+  &__days {
+    margin-top: 30px;
+    display: flex;
+    flex-direction: row;
+    //  justify-content: space-between;
```
Comment (Oskar2301) [inline src/app/shared/components/next-days/next-days.component.scss]: delete

File: src/app/shared/components/next-days/next-days.component.scss [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+      font-weight: 500;
+    }
+
+    &_low {
+      color: #686868;
+    }
+  }
+}
+
+// --------------------------------------
```
Comment (Oskar2301) [inline src/app/shared/components/next-days/next-days.component.scss]: the same

File: src/app/shared/components/next-days/next-days.component.scss [hunk 11]
```diff
@@ -0,0 +1,99 @@
...
+
+@media (max-width: 1150px) {
```
Comment (Oskar2301) [inline src/app/shared/components/next-days/next-days.component.scss]: mixin

File: src/app/shared/components/next-days/next-days.component.ts [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+import * as moment from 'moment';
+
+@Component({
+  selector: 'app-next-days',
+  templateUrl: './next-days.component.html',
+  styleUrls: ['./next-days.component.scss'],
+})
+export class NextDaysComponent implements OnInit {
+  @Input() forecastData: INextDays;
+  public day: any;
```
Comment (Oskar2301) [inline src/app/shared/components/next-days/next-days.component.ts:13]: any? create interfaces for this

File: src/app/shared/components/theme-button/theme-button.component.scss [hunk 13]
```diff
@@ -0,0 +1,20 @@
...
+  background: transparent;
+
+  &__image {
+    width: 40px;
+  }
+}
+
+// --------------------------------------
+
+@media (max-width: 1070px) {
```
Comment (Oskar2301) [inline src/app/shared/components/theme-button/theme-button.component.scss]: mixin

File: src/app/shared/services/weather.service.ts [hunk 14]
```diff
@@ -1,10 +1,97 @@
...
+
+  public getWeatherData(city: string, hour?: string): Observable<IWeatherData> {
+    return this.http.get<any>(
+      `${this.baseUrl}/forecast.json?key=${this.key}&q=${city}&days=3&hour=${hour}&aqi=no&alerts=no`
+    );
+  }
+
+  public async getCities(query: string): Promise<any> {
+    const url = `${this.baseUrl}/search.json?key=${this.key}&q=${query}`;
+    return this.http.get<any>(url).toPromise();
```
Comment (Oskar2301) [inline src/app/shared/services/weather.service.ts]: delete any

File: src/app/shared/services/weather.service.ts [hunk 15]
```diff
@@ -1,10 +1,97 @@
...
+  }
+
+  private isCurrentDay(enumType: any, currentDay: string): boolean {
```
Comment (Oskar2301) [inline src/app/shared/services/weather.service.ts:88]: any

File: src/app/shared/services/weather.service.ts [hunk 16]
```diff
@@ -1,10 +1,97 @@
...
+      return WeathreIconsEnum.Sun;
+    } else if (this.isCurrentDay(SnowEnum, currentDay)) {
+      return WeathreIconsEnum.Snow;
+    } else {
+      return WeathreIconsEnum.Thunder;
+    }
+  }
+
+  public getWeatherData(city: string, hour?: string): Observable<IWeatherData> {
+    return this.http.get<any>(
```
Comment (Oskar2301) [inline src/app/shared/services/weather.service.ts]: any

File: src/app/shared/services/weather.service.ts [hunk 17]
```diff
@@ -1,10 +1,97 @@
...
+  public readonly rainy = WeathreIconsEnum.Rainy;
+  public readonly snow = WeathreIconsEnum.Snow;
+  public readonly thunder = WeathreIconsEnum.Thunder;
+
+  constructor(public http: HttpClient) {}
+
+  public getWeather(city?: string): void {
+    this.getWeatherData(city || this.city).subscribe((value) => {
+      this.weatherData = value;
+      console.log(value);
```
Comment (Oskar2301) [inline src/app/shared/services/weather.service.ts]: delete console

File: src/app/shared/services/weather.service.ts [hunk 18]
```diff
@@ -1,10 +1,97 @@
...
   providedIn: 'root',
 })
 export class WeatherService {
-  private key = '890ac24d69024e8fa11102916231311';
+  public key = '890ac24d69024e8fa11102916231311';
+  public baseUrl = 'http://api.weatherapi.com/v1';
+  public readonly city = 'Vinnytsya';
+  public readonly dayParts: IDayPart[] = dayParts;
+  public readonly nextDays: INextDays[] = nextDays;
+  public weatherData: any;
```
Comment (Oskar2301) [inline src/app/shared/services/weather.service.ts:27]: any

File: src/app/pages/weather-page/weather-page.component.ts [hunk 19]
```diff
@@ -0,0 +1,116 @@
...
+    this.weatherService.getWeather();
+    this.weatherService.triggerRequest.pipe().subscribe(() => {
+      this.initData();
+    });
+    this.searchForm.valueChanges
+      .pipe(debounceTime(300), distinctUntilChanged())
+      .subscribe(async (value: string | null) => {
+        if (value) {
+          const cities = await this.weatherService.getCities(value);
+          console.log(cities);
```
Comment (Oskar2301) [inline src/app/pages/weather-page/weather-page.component.ts]: console?

File: src/app/shared/services/weather.service.ts (hunk 15 above)
Comment (plopdark) [reply src/app/shared/services/weather.service.ts:88]: idk on what change it

File: src/app/shared/services/weather.service.ts (hunk 18 above)
Comment (plopdark) [reply src/app/shared/services/weather.service.ts:27]: same,different type of data

File: src/app/shared/components/next-days/next-days.component.ts (hunk 12 above)
Comment (plopdark) [reply src/app/shared/components/next-days/next-days.component.ts:13]: can't change because i recieve different type of data

**Pull request merged**